src_c/_pygame.h

```diff
@@ -42,6 +42,19 @@
 
 #include <SDL.h>
 
+/* DictProxy is useful for event posting with an arbitrary dict. Maintains
+ * state of number of events on queue and whether the owner of this struct
+ * wants this dict freed. This DictProxy is only to be freed when there are no
+ * more instances of this DictProxy on the event queue. Access to this is
+ * safeguarded with a per-proxy spinlock, which is more optiminal than having
+ * to hold GIL in case of event timers */
+typedef struct _pgEventDictProxy {
+    PyObject *dict;
+    SDL_SpinLock lock;
+    int num_on_queue;
+    Uint8 is_freed;
```

Comment:
Looking at the code, I feel that the name `is_freed` is not helping me understand the purpose.

As far as I can gather, it is basically `free_when_queue_counter_is_zero`... but as far as names go, it has its own problem.  Maybe a `proxy_source_is_closed` if the argument is we should free it because the proxy will never be emitted in an event again.

Reply:
Maybe, I am also missing a comment describing when `num_on_queue` would be `>0` but `is_freed` would be true.  I can see there is code for it but mentally wrapping my head around when this can happen in practice is a different matter.